samples/nrf5340/netboot/src/main.c

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr/types.h>
+#include <sys/printk.h>
+#include <pm_config.h>
+#include <fw_info.h>
+#include <fprotect.h>
+#include <bl_storage.h>
+#include <bl_boot.h>
+#include <bl_validation.h>
+#include <dfu/pcd.h>
+#include <device.h>
+
+#define FLASH_NAME DT_CHOSEN_ZEPHYR_FLASH_CONTROLLER_LABEL
+
+void main(void)
+{
+	int err = fprotect_area(PM_B0N_ADDRESS, PM_B0N_SIZE);
+	struct device *fdev = device_get_binding(FLASH_NAME);
+
+	if (err) {
+		printk("Failed to protect b0n flash, cancel startup\n\r");
+		goto failure;
+	}
+
+	uint32_t s0_addr = s0_address_read();
+	bool valid = false;
+
+	if (pcd_fw_copy_status_get() == PCD_STATUS_COPY) {
+		/* First we validate the data where the PCD CMD tells
+		 * us that we can find it.
+		 */
+		const void *update_addr = pcd_cmd_data_ptr_get();
+
+		if (update_addr == NULL) {
```

Comment:
IMO this check is not needed. NULL is not something that function returns, and the validation will fail anyways if NULL is returned.